source/extensions/filters/network/redis_proxy/router_impl.cc

```diff
@@ -3,10 +3,44 @@
 #include "envoy/extensions/filters/network/redis_proxy/v3/redis_proxy.pb.h"
 #include "envoy/type/v3/percent.pb.h"
 
+#include "source/common/formatter/substitution_formatter.h"
+#include "source/common/http/header_map_impl.h"
+
+#include "absl/strings/str_replace.h"
+#include "iostream"
+
 namespace Envoy {
 namespace Extensions {
 namespace NetworkFilters {
 namespace RedisProxy {
+namespace {
+void formatKey(Network::ReadFilterCallbacks* callbacks, PrefixSharedPtr prefix_shared_ptr,
+               std::string& key, std::string redis_key_formatter_command) {
```

Comment:
It may be simpler to make the function a member function of PrefixRoutes class, and also pass the key formatter instead of prefix_shared_ptr

void PrefixRoutes::formatKey(std::string& key, const std::string& redis_key_formatter) {
 ...
}

Please also add comments about the input parameters and what to expect on the function output, for example:
https://github.com/envoyproxy/envoy/blob/f179528d6fadc4f3357d59ee0e9ffa9307f4c8ee/test/extensions/filters/network/redis_proxy/redis_proxy_integration_test.cc#L375-L379

Reply:
done.

Reply:
Looks great, thanks for addressing the comment!